File: MyMelvorBlazor/MyMelvorBlazor/Models/ItemClass.cs
namespace MyMelvorBlazor.Models
{
	public class ItemClass
	{
		public ItemId? Id { get; set; } = null;
		public string Name { get; set; } = "";
		public string Description { get; set; } = "";

		public ItemClass(ItemId id, string name, string descritpion)
		{
			Id = id;
			Name = name;
			Description = descritpion;
		}
	}
}

File: MyMelvorTS/mymelvorts.client/src/classes/ItemClass.ts
import type ItemType from "./ItemType"

export default class ItemClass {
    Id: number = -1
    Name: string = "";
    Description: string = ""
    Type: ItemType

    constructor(id: number, name: string, description: string, itemType: ItemType) {
        this.Id = id
        this.Name = name
        this.Description = description
        this.Type = itemType
    }
}